src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -476,8 +487,9 @@ private static unsafe int SysReceiveMessageFrom(SafeSocketHandle socket, SocketF
             Debug.Assert(socketAddress != null, "Expected non-null socketAddress");
 
             int buffersCount = buffers.Count;
-            var handles = new GCHandle[buffersCount];
-            var iovecs = new Interop.Sys.IOVector[buffersCount];
+            bool allocOnStack = buffersCount <= IovStackThreshold;
```

Comment:
Do we have tests that'll exercise below / above this threshold?  (Hopefully we do, since we're already doing such a switch on Windows... but it uses a different threshold value.)

Reply:
SendRecvIovMaxTcp_Success uses 2400 iovs.

https://github.com/dotnet/runtime/blob/1705e687a8332a3cf119968eec6ed61b98ceff2a/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs#L1137-L1144